src/oneD/Sim1D.cpp

```
@@ -724,6 +724,80 @@ double Sim1D::fixedTemperatureLocation()
     return z_fixed;
 }
 
+void Sim1D::setLeftControlPoint(double temperature)
+{
+    double epsilon = 1e-3; // Precision threshold for being 'equal' to a temperature
+    for (size_t n = 0; n < nDomains(); n++) {
+        Domain1D& d = domain(n);
+
+        // Skip if the domain type doesn't match
+        if (d.domainType() != "axisymmetric-flow") {
+            continue;
+        }
+
+        StFlow* d_axis = dynamic_cast<StFlow*>(&domain(n));
+        size_t np = d_axis->nPoints();
+
+        // Skip if none of the control is enabled
+        if (!d_axis->twoPointControlEnabled()) {
+            continue;
+        }
+
+        for (size_t m = 0; m < np-1; m++) {
+            // Check if the absolute difference between the two temperatures is less than epsilon
+            if (std::abs(value(n,2,m) - temperature) < epsilon) {
+                d_axis->m_zLeft = d_axis->grid(m);
+                d_axis->m_tLeft = value(n,2,m);
+                return;
+            }
+
+            if ((value(n,2,m) - temperature) * (value(n,2,m+1) - temperature) < 0.0) {
+                d_axis->m_zLeft = d_axis->grid(m+1);
+                d_axis->m_tLeft = value(n,2,m+1);
+                return;
+            }
+        }
+    }
+
+}
```

I think if you end up here, at the end of the loop over the domains, without having set a control point, it should raise an exception -- either two point control isn't enabled for any domain, or the specified temperature is outside the range of temperatures present.

---

I placed a check for both conditions.